src/scene/scene.js

```diff
@@ -626,14 +626,23 @@ class Scene extends EventHandler {
      */
     set skyboxRotation(value) {
         if (!this._skyboxRotation.equals(value)) {
+            const prevIdentity = this._skyboxRotation.equals(Quat.IDENTITY);
+            const newIdentity = value.equals(Quat.IDENTITY);
+
             this._skyboxRotation.copy(value);
-            if (value.equals(Quat.IDENTITY)) {
+
+            if (newIdentity) {
                 this._skyboxRotationMat3.setIdentity();
             } else {
                 this._skyboxRotationMat4.setTRS(Vec3.ZERO, value, Vec3.ONE);
                 this._skyboxRotationMat4.invertTo3x3(this._skyboxRotationMat3);
             }
-            this._resetSky();
+
+            // only reset sky / rebuild scene shaders if rotation changed from identity, to include
+            // support for skybox rotation in shaders
+            if (prevIdentity && !newIdentity) {
```

Comment:
This should either do `prevIdentity !== newIdentity` or scene should store a flag `skyboxHasRotation` which tracks first time rotation is set (and then doesn't reset again).

Reply:
I don't want to rebuild shaders if we go from rotation to no rotation.

Reply:
Right, but now you're rebuilding every time you go no->rotation. I think you want a flag that says 'we've compiled rotation' and only do that once.

Reply:
ok this is done, thanks